DOC Added dropdowns to 6.2 feature-extraction

Kishan-Ved opened the pull request:
Reference Issues/PRs
Add dropdowns to submodule 6.2 Feature Extraction regarding https://github.com/scikit-learn/scikit-learn/issues/26617.
Folded:

6.2.2.1. Implementation details (Cross referencing taken care of by adding it inside the dropdown)
6.2.3.9. Performing out-of-core scaling with HashingVectorizer

SIDE NOTE:
I am on summer vacation, I can work extensively on this and more contributions provided my PR is accepted, so I know I'm on the right track as this is my first time contributing to open source :) .

Kishan-Ved committed:
Added dropdowns to 6.2 feature-extraction

Kishan-Ved commented:
@betatim Hello, can you please help!, this is my first open source contribution :)

ArturoAmorQ commented on doc/modules/feature_extraction.rst:
```diff
@@ -223,16 +224,17 @@ Since a simple modulo is used to transform the hash function to a column index,
 it is advisable to use a power of two as the ``n_features`` parameter;
 otherwise the features will not be mapped evenly to the columns.
 
+.. topic:: References:
+  * `MurmurHash3 <https://github.com/aappleby/smhasher>`_.
```
The reference is not being rendered because it needs an extra space:

```suggestion
.. topic:: References:

  * `MurmurHash3 <https://github.com/aappleby/smhasher>`_.
```

Kishan-Ved replied:
I have fixed this, thank you. I have also added the 4 dropdowns that you have mentioned, please review the PR for a merge.

ArturoAmorQ reviewed (commented):
Thanks for the PR @Kishan-Ved!

I think this section can still make use of more dropdowns, e.g.:
- a dropdown for "using stop words" in 6.2.3.3
- a dropdown for "worked-out example" in 6.2.3.4
- a dropdown for "troubleshooting decoding text" in 6.2.3.5
- a dropdown for "tips and tricks" in 6.2.3.9

Kishan-Ved committed:
Added dropdowns to 6.2 feature_extraction

Updated based on the review by @ArturoAmorQ.
Added:
- A dropdown for "using stop words" in 6.2.3.3
- A dropdown for "worked-out example" in 6.2.3.4
- A dropdown for "troubleshooting decoding text" in 6.2.3.5
- A dropdown for "tips and tricks" in 6.2.3.9

ArturoAmorQ commented on doc/modules/feature_extraction.rst:
```diff
@@ -490,6 +497,10 @@ class::
 <text_feature_extraction_ref>` for the details on all the parameters.
 
+|details-start|
+Example
```
Maybe we can be a bit more explicit here:

```suggestion
Numeric example of a tf-idf matrix
```

Kishan-Ved replied:
I've done it, thanks for the suggestion! Kindly check.

Kishan-Ved replied:
Good to go? @ArturoAmorQ

ArturoAmorQ reviewed (approved):
Just a tweak. Otherwise LGTM!

Kishan-Ved committed:
Added dropdowns to 6.2 feature_extraction

Changed line 501

ArturoAmorQ commented:
> Good to go? @ArturoAmorQ

On my side, yes. We need another approval, though.

raph333 reviewed (approved):
The changes look good to me. Thanks for your work!

But since I'm a newbie here, we still need another review :-)

ArturoAmorQ committed:
Merge branch 'main' into dropdowns-feature-extraction

ArturoAmorQ commented on doc/modules/feature_extraction.rst:
```diff
@@ -490,6 +497,10 @@ class::
 <text_feature_extraction_ref>` for the details on all the parameters.
 
+|details-start|
+Numeric example of a tf-idf matrix
```
```suggestion
**Numeric example of a tf-idf matrix**
```

ArturoAmorQ committed:
Update doc/modules/feature_extraction.rst

ArturoAmorQ commented:
> We need another approval, though.

According to the [Decision making process](https://scikit-learn.org/dev/governance.html#decision-making-process) I think we can consider this as a "minor documentation change" and should be good to go with my approval.

Long story short, I am merging this one. Thanks again for your work and patience @Kishan-Ved :)
